src/Analyzers/Core/Analyzers/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnalyzer.cs

```diff
@@ -65,20 +67,13 @@ private void AnalyzeInterpolation(OperationAnalysisContext context)
                 return;
             }
 
-            context.ReportDiagnostic(DiagnosticHelper.Create(
-                UnnecessaryWithSuggestionDescriptor,
+            context.ReportDiagnostic(DiagnosticHelper.CreateWithLocationTags(
+                Descriptor,
                 unnecessaryLocations.First(),
                 option.Notification.Severity,
-                additionalLocations: ImmutableArray.Create(interpolation.Syntax.GetLocation()),
-                properties: null));
-
-            // We start at 1 because the 0th element was used above to make the main diagnostic descriptor.
-            // All the rest are used to just fade out the correct portions of the user's code.
```

Comment:
i like that comment :)  definitely would still be helpful even with teh new code.

Reply:
➡️ Now resolved